src/entities/MovieForm/ui/MovieForm.jsx

```diff
@@ -1,106 +1,183 @@
-import React, { useState } from "react";
+import React from "react";
+import { useOutletContext } from "react-router";
 import "./MovieForm.css";
 import { genresList } from "../../../entities/SelectTabs/selectTabsMock";
 import { getMovie } from "../lib/getMovie";
+import { useForm, Controller } from "react-hook-form";
 
-export const MovieForm = ({ initialMovie, onFormSubmit }) => {
-  const [movie, setMovie] = useState(getMovie(initialMovie));
+export const MovieForm = ({ initialMovie }) => {
+  const { movie } = useOutletContext() ?? {initialMovie};
+ 
+  const { handleSubmit, control, reset } = useForm({
+    defaultValues: getMovie(movie),
+  });
 
-  const handleFieldChange = (event) => {
-    setMovie((prevMovie) => ({
-      ...prevMovie,
-      [event.target.name]: event.target.value,
-    }));
+  const onFormSubmit = async (data) => {
+    try {
+      let requestOptions;
+      if (movie) {
+        requestOptions = {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(data),
+        };
+      } else {
+        requestOptions = {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(data),
+        };
+      }
+      await fetch("http://localhost:4000", requestOptions);
```

Comment:
can be moved to a variable `http://localhost:4000`

Reply:
done